frontend/packages/console-dynamic-plugin-sdk/README.md

```diff
@@ -186,20 +189,47 @@ Console application uses [Content Security Policy](https://developer.mozilla.org
 includes the document origin `'self'` and Console webpack dev server when running off-cluster.
 
 All dynamic plugin assets _should_ be loaded using `/api/plugins/<plugin-name>` Bridge endpoint which
-matches the `'self'` CSP source of Console application.
+matches the `'self'` CSP source for all Console assets served via Bridge.
 
-See `cspSources` and `cspDirectives` in
-[`pkg/server/server.go`](https://github.com/openshift/console/blob/master/pkg/server/server.go)
+Refer to `BuildCSPDirectives` function in
+[`pkg/utils/utils.go`](https://github.com/openshift/console/blob/master/pkg/utils/utils.go)
 for details on the current Console CSP implementation.
 
+Starting with Console 4.18, dynamic plugins may extend existing Console application CSP directives
+using `spec.contentSecurityPolicy` field of `ConsolePlugin` resource, for example:
+
+```yaml
+apiVersion: console.openshift.io/v1
+kind: ConsolePlugin
+metadata:
+  name: cron-tab
+spec:
+  displayName: 'Cron Tab'
+  contentSecurityPolicy:
+  - directive: 'ScriptSrc'
+    values:
+    - 'https://example1.com/'
+    - 'https://example2.com/'
+```
+
+Refer to [Dynamic Plugins feature page][console-doc-feature-page] section on Content Security Policy
+for more details.
+
 ### Changes in Console CSP
 
-This section documents notable changes in the Console Content Security Policy.
+This section documents notable changes in the Console Content Security Policy implementation.
 
 #### Console 4.18.x
 
-Console CSP is deployed in report-only mode. CSP violations will be logged in the browser console
-but the associated CSP directives will not be enforced.
+Console deploys CSP in report-only mode; CSP violations will be logged in the browser console
+and CSP violation data may be reported through telemetry service in production deployments.
+
+In this version, the CSP feature must be enabled via feature gate in order for Console to collect
+and process CSP violation data.
```

Comment:
```suggestion
In a future release, console will begin enforcing CSP. Consider preparing your plugin now. To test your plugin with CSP, enable the `ConsolePluginContentSecurityPolicy` feature gate on a test cluster. This feature gate should **not** be enabled on a production cluster. Enabling the feature gate also allows you to set `spec. contentSecurityPolicy` in you `ConsolePlugin` resource to declare any fetch directives that your plugin needs.
```

@opayne1 Have we added any information about this to the 4.18 release notes?

Reply:
Makes sense, I'll add that to the doc.